tests/click_tests/device_menu/test_wipe_code.py

```diff
@@ -0,0 +1,312 @@
+
+
+@pytest.mark.setup_client(pin=None)
+def test_wipe_code_setup_pin_unset(device_handler: "BackgroundDeviceHandler"):
+
+    # Device doesn't have the pin set, the wipe code button is not accessible
+    with pytest.raises(
+        AssertionError,
+        match=rf"['\"]{re.escape(TR.translate(WIPE_CODE_TITLE))}['\"] in",
+    ):
+        prepare_wipe_dialogue(device_handler, Situation.SETUP)
+
+
+@pytest.mark.setup_client(pin=PIN4)
+def test_wipe_code_setup(device_handler: "BackgroundDeviceHandler"):
+    enter_pin(device_handler)
+
+    debug = device_handler.debuglink()
+    features = device_handler.features()
+
+    assert features.wipe_code_protection is False
+
+    prepare_wipe_dialogue(device_handler, Situation.SETUP)
+
+    change_wipe_code(debug, wipe_code=WIPE_CODE)
+
+    assert_device_screen(debug, Menu.SECURITY)
+    close_device_menu(debug)
+
+    features = device_handler.features()
+    assert features.wipe_code_protection is True
+
+
+@pytest.mark.setup_client(pin=PIN4)
+def test_wipe_code_same_as_pin(device_handler: "BackgroundDeviceHandler"):
+    enter_pin(device_handler)
+
+    debug = device_handler.debuglink()
+    prepare_wipe_dialogue(device_handler, Situation.SETUP)
+
+    # Intro screen
+    go_next(debug)
+    # Enter pin to authorize
+    _input_see_confirm(debug, PIN4)
+    # Input wipe code same as pin
+    _input_see_confirm(debug, PIN4)
+    # Try again
+    go_next(debug)
+    _enter_two_times(debug, WIPE_CODE, WIPE_CODE)
+    # Confirmation screen
+    go_next(debug)
+
+    assert_device_screen(debug, Menu.SECURITY)
+    close_device_menu(debug)
+
+    features = device_handler.features()
+    assert features.wipe_code_protection is True
+
+
+@pytest.mark.setup_client(pin=PIN4)
+def test_wipe_code_cancel_intro(device_handler: "BackgroundDeviceHandler"):
+    enter_pin(device_handler)
+
+    debug = device_handler.debuglink()
+    features = device_handler.features()
+
+    assert features.wipe_code_protection is False
+
+    prepare_wipe_dialogue(device_handler, Situation.SETUP)
+
+    # Go to menu in the intro screen
+    debug.click(debug.screen_buttons.menu())
+    # Press cancel
+    debug.button_actions.navigate_to_menu_item(0)
+    # Confirm cancel
+    go_next(debug)
+
+    assert_device_screen(debug, Menu.SECURITY)
+    close_device_menu(debug)
+
+    features = device_handler.features()
+    assert features.wipe_code_protection is False
+
+
+@pytest.mark.setup_client(pin=PIN4)
+def test_wipe_code_cancel_keyboard(device_handler: "BackgroundDeviceHandler"):
+    enter_pin(device_handler)
+
+    debug = device_handler.debuglink()
+    features = device_handler.features()
+
+    assert features.wipe_code_protection is False
+
+    prepare_wipe_dialogue(device_handler, Situation.SETUP)
+
+    # Intro screen
+    go_next(debug)
+    # Enter pin to authorize
+    _input_see_confirm(debug, PIN4)
+    # Enter wipe code, then cancel
+    _input_pin(debug, PIN1)
```

Comment:
nit: maybe rename to `_input_code`? (since it's also used for wipe code)
Also for the rest of the PIN-related helper functions below...